### PR TITLE
Add a way to disable non error request logging

### PR DESCRIPTION
**Pull request by iwismer**
This makes changes to the request logging interceptor in order to allow disabling logging of non-error requests completely.

Currently, the logging interceptor does not allow you to set the rate limit to 0 as 0 is used to specify no limit. I didn't want to break backwards compatility to change it from 0 to -1 for no limit and 0 for no logging. Instead I've added a new config boolean that can be used to set it to not log for non-error responses.

The reason that I've added this is that a rate limit of 1 is too high for high volume APIs when there are a large number of pods handling requests. The rate limiting is done on a per pod basis, so a rate limit of 1/second with 100 pods will result in about 8.5million logs per day. I'd like a way to just turn these off!

### CODE REVIEW

**Review by becassel — APPROVED** [review]

**Comment by becassel** [comment]
Looks good to me, one strong potential alternative might be to use an enum logging level config (i.e. `ALL`, and `ERROR_ONLY`), but otherwise this accomplishes what we want, would defer to the Misk maintainers about their preferences (if any).

File: misk-actions/src/main/kotlin/misk/web/interceptors/LogRequestResponse.kt
Comment (becassel) [inline misk-actions/src/main/kotlin/misk/web/interceptors/LogRequestResponse.kt:40]: Consider using an enum instead, i.e. `val loggingLevel: LoggingLevel = LoggingLevel.ALL`, and then have another level like `ERROR_ONLY` which has your equivalent behaviour.

Comment (iwismer) [reply misk-actions/src/main/kotlin/misk/web/interceptors/LogRequestResponse.kt]: Done!

File: misk-actions/src/main/kotlin/misk/web/interceptors/LogRequestResponse.kt
Comment (mhickman) [inline misk-actions/src/main/kotlin/misk/web/interceptors/LogRequestResponse.kt:44]: what do you think of naming this `RequestLoggingMode` ?

Comment (iwismer) [reply misk-actions/src/main/kotlin/misk/web/interceptors/LogRequestResponse.kt:44]: Naming this was hard, I'm happy to rename it to that.

**Review by frojasg — APPROVED** [review]
LGTM

**Pull request merged**
